script.js

```diff
@@ -1,4 +1,229 @@
+var gameMode = "";
+var userWinCount = 0;
+var computerWinCount = 0;
+var drawCount = 0;
+var userName = "";
+var koreanWinner = "";
+
+var randomSPS = function () {
+  randomDecimal = Math.random();
+  randomInteger0to2 = Math.floor(randomDecimal * 3);
+  RandomInterger1to3 = randomInteger0to2 + 1;
+  if (RandomInterger1to3 == 1) {
+    computerChoiceMsg = "The computer chose Scissors!";
+    return "Scissors";
+  }
+  if (RandomInterger1to3 == 2) {
+    computerChoiceMsg = "The computer chose Paper!";
+    return "Paper";
+  }
+  if (RandomInterger1to3 == 3) {
+    computerChoiceMsg = "The computer chose Stone!";
+    return "Stone";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  computerChoice = randomSPS();
+  var yourChoiceMsg = "You chose " + input + "!";
+  // Enter username
+  if (userName == "") {
+    userName = input;
+    if (userName == "") {
+      myOutputValue = "Please enter a valid username and click submit.";
+      return myOutputValue;
```

Comment:
You could check for user input at the beginning of the main function. (line 27) 

Reason being that if the user did not input anything, we would not need to call the randomSPS() function at all.